mcodingbot/plugins/highlights.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+import asyncio
+from collections import defaultdict
+import contextlib
+
+import crescent
+import hikari
+from mcodingbot.utils import Plugin, Context
+from mcodingbot.database.models.user import Word, User
+
+plugin = Plugin()
+highlights_group = crescent.Group("highlights")
+highlights_cache: dict[str, list[hikari.Snowflake]] = defaultdict(list)
+
+
+def _cache_highlight(word: str, *user_ids: hikari.Snowflake) -> None:
+    for user_id in user_ids:
+        highlights_cache[word].append(user_id)
+
+
+def _uncache_highlight(word: str, *user_ids: hikari.Snowflake) -> None:
+    for user_id in user_ids:
+        highlights_cache[word].remove(user_id)
+        # Deletes empty arrays from the cache
+        if not highlights_cache[word]:
+            del highlights_cache[word]
+
+
+@plugin.include
+@highlights_group.child
+@crescent.command(name="create")
+class CreateHighlight:
+    word = crescent.option(str)
+
+    async def callback(self, ctx: Context) -> None:
+        if len(self.word) > 32:
+            await ctx.respond("Highlights can not be longer than 32 characters.")
+            return
+        await User.add_word(self.word, ctx.user.id)
+        await ctx.respond(f'Added "{self.word}" to your highlights.')
+
+        _cache_highlight(self.word, ctx.user.id)
+
+
+@plugin.include
+@highlights_group.child
+@crescent.command(name="delete")
+class DeleteHighlight:
+    word = crescent.option(str)
+
+    async def callback(self, ctx: Context) -> None:
+        if await User.delete_word(self.word, ctx.user.id):
+            await ctx.respond(f'Removed "{self.word}" from your highlights.')
+            _uncache_highlight(self.word, ctx.user.id)
+            return
+
+        await ctx.respond(f'"{self.word}" was not one of your highlights.')
+
+
+@plugin.include
+@highlights_group.child
+@crescent.command(name="list")
+async def list(ctx: Context) -> None:
+    if not await User.exists(user_id=ctx.user.id):
+        await ctx.respond("You do not have any highlights.")
+        return
+
+    user = await User.fetch(user_id=ctx.user.id)
```

Comment:
```py
if not (user := await User.exists(user_id=ctx.user.id)):
    await ctx.respond("You do not have any highlights.")
    return
```

Reply:
I think using the walrus operator when you aren't using the variable inside if the if statement is an antipattern tbh

Reply:
I don't think so, it's how a lot of people use it (sourcery also recommends this pattern)

Reply:
the walrus greatly increases cognitive complexity

Reply:
Then do ```py
user = await User.exists(user_id=ctx.user.id)
if not user:
    await ctx.respond(...)
    return
```

Reply:
thats what im doing. I think its much more readable, and especially much more beginner friendly.